### PR TITLE
Decouple TestVerify into TestValidateBasic and TestVerify

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1189 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by MSevey** [comment]
this is based on #1228 right? can we target that branch to reduce the diff or just make this as draft? there are comments on the other PR that I don't want to duplicate.

**Comment by Manav-Aggarwal** [comment]
> this is based on #1228 right? can we target that branch to reduce the diff or just make this as draft? there are comments on the other PR that I don't want to duplicate.

Yes but since those PRs are not related, I made the PR independent of #1228 so it's not based on it anymore. Thanks!

File: types/signed_header_test.go [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
+func TestSignedHeader(t *testing.T) {
+	// Generate a random signed header
 	trusted, privKey, err := GetRandomSignedHeader()
 	require.NoError(t, err)
+
+	// Wait for a second to ensure time difference before generating the next header
 	time.Sleep(time.Second)
+
+	// Get the next random header
 	untrustedAdj, err := GetNextRandomHeader(trusted, privKey)
```
Comment (MSevey) [inline types/signed_header_test.go:23]: GetNextRandomHeader should just set the time to plus 1 second so that the tests don't need to wait a second.

**Review by MSevey — COMMENTED** [review]
minor test util improvement, otherwise lgtm

**Review by S1nus — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**